test/rules/assertions/checkTagNames.js

```diff
@@ -936,5 +1282,26 @@ export default {
         },
       },
     },
+    {
+      code: `
+        interface WebTwain {
+          /**
+           * Converts the images specified by the indices to base64 synchronously.
+           * @function WebTwain#ConvertToBase64
+           * @returns {Base64Result}
+       
+          ConvertToBase64(): Base64Result;
+          */
+
+          /**
+           * Converts the images specified by the indices to base64 asynchronously.
+           * @function WebTwain#ConvertToBase64
+           * @returns {boolean}
+           */
+          ConvertToBase64(): boolean;
+        }
+      `,
+      parser: require.resolve('@typescript-eslint/parser'),
+    },
```

Comment:
This is extracted from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c8089e91d1be9af95c604ef234180903593a07b3/types/dwt/v13/index.d.ts. `node` is `null` in the lint rule:

```plaintext
TypeError: Cannot read properties of null (reading 'parent')
Occurred while linting /Users/josh/repos/DefinitelyTyped/types/dwt/v13/index.d.ts:23
Rule: "jsdoc/check-tag-names"
    at tagIsRedundantWhenTyped (/Users/josh/repos/eslint-plugin-jsdoc/dist/rules/checkTagNames.js:66:57)
    at checkTagForTypedValidity (/Users/josh/repos/eslint-plugin-jsdoc/dist/rules/checkTagNames.js:97:9)
    at _default.iterateAllJsdocs (/Users/josh/repos/eslint-plugin-jsdoc/dist/rules/checkTagNames.js:113:18)
    ...
```